File: apps/backend-node/src/features/bookings/dto/booking.dto.ts
import type { BookingStatus, BookingType } from './booking-enums.dto';

/** OpenAPI Booking response. */
export interface BookingDto {
  id: string;
  customerId: string;
  roomId: string;
  propertyId: string;
  checkIn: string;
  checkOut: string;
  guestCount: number;
  status: BookingStatus;
  bookingType: BookingType;
  holdExpiresAt: string | null;
  customerNotes: string | null;
  approvedAt: string | null;
  approvedBy: string | null;
  cancelledAt: string | null;
  paymentRetryCount: number;
  createdAt: string;
  updatedAt: string;
}

File: apps/backend-java/src/main/java/com/housingplatform/features/bookings/dto/BookingDto.java
package com.housingplatform.features.bookings.dto;

import com.housingplatform.persistence.enums.BookingStatus;
import com.housingplatform.persistence.enums.BookingType;
import jakarta.validation.constraints.Min;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.UUID;

public record BookingDto(
    UUID id,
    UUID customerId,
    UUID roomId,
    UUID propertyId,
    LocalDate checkIn,
    LocalDate checkOut,
    @Min(1) int guestCount,
    BookingStatus status,
    BookingType bookingType,
    OffsetDateTime holdExpiresAt,
    String customerNotes,
    OffsetDateTime approvedAt,
    UUID approvedBy,
    OffsetDateTime cancelledAt,
    @Min(0) int paymentRetryCount,
    OffsetDateTime createdAt,
    OffsetDateTime updatedAt) {}
